packages/wallets/src/components/WalletListCardDropdown/WalletListCardDropdown.tsx

```diff
@@ -27,66 +39,117 @@ const WalletListCardDropdown: React.FC<TSubscribedBalance> = ({ balance }) => {
         [t]
     );
 
-    useEffect(() => {
-        const selectedWallet = wallets?.find(wallet => wallet.loginid === loginId);
-        if (selectedWallet) {
-            const selectedTextWidth = generateTitleText(selectedWallet).length;
-            setInputWidth(`${selectedTextWidth * 10 - 20}px`);
+    const walletList: WalletList = useMemo(() => {
+        return (
+            wallets
+                ?.filter(wallet => !wallet.is_virtual)
+                .map(wallet => ({
+                    currency: wallet.currency,
+                    currencyConfig: wallet.currency_config,
+                    loginid: wallet.loginid,
+                    text: generateTitleText(wallet),
+                })) ?? []
+        );
+    }, [generateTitleText, wallets]);
+
+    useOnClickOutside(dropdownRef, () => {
+        setIsOpen(false);
+    });
+
+    useEventListener('keydown', (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+            setIsOpen(false);
```

Comment:
Added event listeners for clicking outside of dropdown and 'Esc' button to close the dropdown list